### PR TITLE
[ET-1925] Add information to telemetry

### PR DESCRIPTION
**Pull request by redscar**
### 🎫 Ticket

[ET-1925] 

### 🎥 Artifacts 

Screenshot 2024-01-02 at 11-33-43 Site Health - Info ‹ tecdev — WordPress

### ✔️ Checklist
- [ ] I've included a changelog entry in the readme.txt file. 
- [ ] My code is tested. 
- [ ] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-1925]: https://stellarwp.atlassian.net/browse/ET-1925?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by estebanhdzc — APPROVED** [review]

**Comment by estebanhdzc** [comment]
@redscar everything looks good to me. I don't know if you need to resolve the PHPCS bot suggestions before merge, the yoda conditions and the spacing issue

File: src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+	 * @return string Formatted average price.
+	 * */
+	private function get_tickets_commerce_average_order_total(): string {
+		// @todo redscar This logic may be incorrect.
+		$tickets_commerce_ticket_prices = tribe( Tickets_Repository::class )->per_page( -1 )->pluck( 'price' );
+		$total                          = 0;
+		$count                          = 0;
+
+		foreach ( $tickets_commerce_ticket_prices as $price ) {
+			if ( $price === 'Free' || $price === '' || $price === null ) {
```
[the remark this answers is not in the document]
Comment (codingmusician) [reply src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php:135]:
```suggestion
			if ( 'Free' === $price || '' === $price || null === $price ) {
```

File: src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+				if ( isset( $matches[0] ) ) {
+					// Convert to a standard number format (replace comma with period).
+					$number = floatval(
+						str_replace(
+							',',
+							'.',
+							$matches[0]
+						)
+					);
+					$total  += $number;
```
[the remark this answers is not in the document]
Comment (codingmusician) [reply src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php:154]:
```suggestion
					$total += $number;
```

File: src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+	 * Determines if Tickets Commerce is in test mode.
+	 *
+	 * @return string 'True' if Tickets Commerce is in test mode, 'False' otherwise.
+	 */
+	private function is_tickets_commerce_test_mode(): string {
+		return $this->get_boolean_string(
+			tribe_get_option(
+				'tickets-commerce-test-mode',
+				false
+			)
```
Comment (codingmusician) [inline src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php:185]: Did a quick codebase search and could find `tickets-commerce-test-mode` as an option. I did find `tickets-commerce-sandbox`, though. Should this be changed?
```suggestion
			tribe_get_option(
				'tickets-commerce-sandbox',
				false
			)
```

Comment (codingmusician) [reply src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php:185]: Also, is there a reason we're not using the variable for all these options (example: `TEC\Tickets\Commerce\Settings::$option_sandbox`)?

File: src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+
+	/**
+	 * Retrieves the currency code set in Tickets Commerce.
+	 *
+	 * @return string The currency code.
+	 */
+	private function get_tickets_commerce_currency_code(): string {
+		return tribe_get_option(
+			'tickets-commerce-currency-code'
+		);
```
Comment (codingmusician) [inline src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php:215]: Should we be including the fallback/defaults for all these options?
```suggestion
		return tribe_get_option(
			'tickets-commerce-currency-code',
			'USD'
		);
```
Reference: `TEC\Tickets\Commerce\Utils\Currency::$currency_code_fallback`

File: src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+	/**
+	 * Gets the currency position setting from Tickets Commerce.
+	 *
+	 * @return string The currency position.
+	 */
+	private function get_tickets_commerce_currency_position(): string {
+		return tribe_get_option(
+			'tickets-commerce-currency-position'
+		);
+	}
```
Comment (codingmusician) [inline src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php:227]: Default should be `prefix`:
```suggestion
	private function get_tickets_commerce_currency_position(): string {
		return tribe_get_option(
			'tickets-commerce-currency-position',
			'prefix'
		);
	}
```

File: src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+
+	/**
+	 * Obtains the decimal separator setting from Tickets Commerce.
+	 *
+	 * @return string The decimal separator.
+	 */
+	private function get_tickets_commerce_decimal_separator(): string {
+		return tribe_get_option(
+			'tickets-commerce-currency-decimal-separator'
+		);
```
Comment (codingmusician) [inline src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php:238]: Default is `.` (`TEC/Tickets/Commerce/Utils/Currency::$currency_code_decimal_separator`):
```suggestion
		return tribe_get_option(
			'tickets-commerce-currency-decimal-separator',
			'.'
		);
```

File: src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+
+	/**
+	 * Retrieves the thousands separator setting from Tickets Commerce.
+	 *
+	 * @return string The thousands separator.
+	 */
+	private function get_tickets_commerce_thousands_separator(): string {
+		return tribe_get_option(
+			'tickets-commerce-currency-thousands-separator'
+		);
```
Comment (codingmusician) [inline src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php:249]: Default is `,` (`TEC/Tickets/Commerce/Utils/Currency::$currency_code_thousands_separator`):
```suggestion
		return tribe_get_option(
			'tickets-commerce-currency-thousands-separator',
			','
		);
```

File: src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+
+	/**
+	 * Gets the number of decimals setting from Tickets Commerce.
+	 *
+	 * @return string The number of decimals.
+	 */
+	private function get_tickets_commerce_number_of_decimals(): string {
+		return tribe_get_option(
+			'tickets-commerce-currency-number-of-decimals'
+		);
```
Comment (codingmusician) [inline src/Tickets/Site_Health/Subsections/Features/Tickets_Commerce_Subsection.php:260]: Default is `2` (`TEC/Tickets/Commerce/Utils/Currency::$currency_code_number_of_decimals`):
```suggestion
		return tribe_get_option(
			'tickets-commerce-currency-number-of-decimals',
			'2'
		);
```

File: src/Tickets/Site_Health/Subsections/Plugins/Plugin_Data_Subsection.php [hunk 9]
```diff
@@ -0,0 +1,654 @@
...
+			$total_and_count['count']
+		);
+
+		return [
+			'formatted_max_price'     => number_format(
+				$max_price,
+				2,
+				'.',
+				','
+			),
```
Comment (codingmusician) [inline src/Tickets/Site_Health/Subsections/Plugins/Plugin_Data_Subsection.php:420]: Everything looks really good! I just have a question about the number format. Should we be pulling the decimal point amount and separators (thousands and decimal) from their configuration?

Comment (redscar) [reply src/Tickets/Site_Health/Subsections/Plugins/Plugin_Data_Subsection.php]: Good point. I updated it to use the currency util instead.

File: src/Tickets/Site_Health/Subsections/Plugins/Plugin_Data_Subsection.php [hunk 10]
```diff
@@ -0,0 +1,643 @@
...
+		);
+		return isset( $matches[0] ) ? floatval(
+			str_replace(
+				',',
+				'.',
+				$matches[0]
+			)
+		) : null;
+	}
+
```
Comment (codingmusician) [inline src/Tickets/Site_Health/Subsections/Plugins/Plugin_Data_Subsection.php:587]: I missed this, before. Will the thousands separated ever be included in the `$price` parameter? If so, won't this turn `1,000.25` into `1.000.25`? Do we need to handle prices like `1.000,25`, too?

**Review by codingmusician — APPROVED** [review]

**Pull request merged**